consensus/src/simplex/actors/resolver/actor.rs

```diff
@@ -195,26 +202,27 @@ impl<E: Clock + GClock + Rng, C: Scheme, H: Hasher, S: Supervisor<Index = View>>
                 return;
             }
 
-            // Select notarizations by ascending height rather than preferring all notarizations or all nullifications
-            //
-            // It is possible we may have requested notarizations and nullifications for the same view (and only one may
-            // exist). We should try to fetch both before trying to fetch the next view, or we may never ask for an existing
-            // notarization or nullification.
+            // We assume nothing about the usefulness (or existence) of any given entry, so we shuffle
+            // the iterator to ensure we eventually try to fetch everything requested.
+            let mut entries = self
+                .required
+                .iter()
+                .filter(|entry| !self.inflight.contains(entry))
+                .collect::<Vec<_>>();
+            if entries.is_empty() {
+                return;
+            }
+            entries.shuffle(&mut self.runtime);
```

Comment:
Are we worried at all about the case where this vector can be large?
In that case we may prefer sampling the `self.required` values without putting them all into a vector and shuffling?

Reply:
The "best" way to do this is to use one of `choose`, `choose_stable`, `choose_multiple` AFAICT: https://docs.rs/rand/latest/rand/seq/trait.IteratorRandom.html#method.choose

I looked at this when working on this PR but backed away because the most "correct" option (`choose_multiple`) says "Although the elements are selected randomly, the order of elements in the buffer is neither stable nor fully random. If random ordering is desired, shuffle the result." I'll give it another pass.

Reply:
I ended up digging into the code here and this actually works pretty well for us (not "super" efficient but more efficient).

Reply:
The responder will iterate in the order they receive the items (so it may result in some favoritism but it'll be fine).

Reply:
For posterity's sake, this still iterates over all items but may just perform many less copies.